### PR TITLE
feat: add macos support

### PR DESCRIPTION
**Pull request by bobbyg603**
### Description

* Adds macos 12+ support
* Adds cross-platform test framework

### Checklist

- [x] Tested manually
- [x] Unit tests pass with no errors or warnings
- [x] Documentation updated (if applicable)
- [x] Reviewed by at least 1 other contributor

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   ],
   "scripts": {
     "build": "tsc && copyfiles bin/LICENSE bin/linux/* dist",
-    "test": "ts-node ./bin/index.ts ./test/my-ubuntu-crasher",
+    "test": "ts-node ./node_modules/.bin/jasmine",
```
Comment (csmith0651) [inline package.json:14]: [NAB] (not a blocker) Just curious what's this for? Why ohhh...it's a testing framework. ok.

File: spec/bin.spec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        expect(stdout).toContain('MODULE mac arm64 B7EBC13028993DE9A52CCAC5C2BB11120 addon.node');
+    });
+
+    pit('linux', 'should dump symbols for linux', async () => {
+        const { stdout, stderr } = await exec(`ts-node ${testScript} ${binary}`);
+
+        expect(stderr).not.toBeDefined();
+        expect(stdout).toContain('MODULE Linux x86_64 C02619C863B6901D53DDD444CED8D6090 802c7064eb1eb5e6b839d554c4626ab3');
+    });
+});
```
Comment (csmith0651) [inline spec/bin.spec.ts:24]: [PRAISE] very nice.

File: spec/support/platform-spec.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const { platform } = process;
+
+export function pit(nodePlatform: string, description: string, test: () => Promise<void> | (() => void)) {
```
Comment (csmith0651) [inline spec/support/platform-spec.ts:3]: [NAB] 'pit' stands for? just curious.

File: src/dump-syms.ts [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
 import * as child_process from 'child_process';
 const exec = util.promisify(child_process.exec);
 
 export async function dumpSyms(filePath: string): Promise<{ stdout: string, stderr: string}> {
     const { platform } = process;
-    const isLinux = platform === 'linux';
-    
-    if (!isLinux) {
+    const dumpSymsPath = path.join(__dirname, `../bin/${platform}/dump_syms`);
+    const supported = await exists(dumpSymsPath);
```
Comment (csmith0651) [inline src/dump-syms.ts]: [QUESTION] any reason you don't use I think....fs.existsSync ?

**Review by csmith0651 — APPROVED** [review]
Since you already have this PR open I would just finish it off with our shared ideas:
- remove set of allowed extensions.
- use the existence of a directory to confirm support of the platform.

Other things that require a bit more detail we can leave as a todo, e.g. figuring if there's a better way to support status being non-zero (you just check stderr if there's data which isn't necessarily indicative of status != 0).

File: spec/bin.spec.ts (hunk 2 above)
Comment (bobbyg603) [reply spec/bin.spec.ts:24]: Ty!

File: src/dump-syms.ts (hunk 4 above)
Comment (bobbyg603) [reply src/dump-syms.ts]: I think existsSync has been deprecated?

File: spec/support/platform-spec.ts (hunk 3 above)
Comment (bobbyg603) [reply spec/support/platform-spec.ts:3]: Platform-it I guess? Open to suggestions! I’ve seen this pattern before but I forget where…

File: src/dump-syms.ts (hunk 4 above)
Comment (bobbyg603) [reply src/dump-syms.ts]: ah! my mistake, it seems only fs.exists has been deprecated:

```
fs.exists() is deprecated, but fs.existsSync() is not
```

from https://nodejs.org/api/fs.html#fsexistssyncpath

File: package.json (hunk 1 above)
Comment (bobbyg603) [reply package.json:14]: 👍

**Comment by bobbyg603** [comment]
> Since you already have this PR open I would just finish it off with our shared ideas:
> 
> * remove set of allowed extensions.
> * use the existence of a directory to confirm support of the platform.
> 
> Other things that require a bit more detail we can leave as a todo, e.g. figuring if there's a better way to support status being non-zero (you just check stderr if there's data which isn't necessarily indicative of status != 0).

I spent half an hour or so trying to figure out if there was a better way to support non-zero return codes and did not find one.

File: spec/support/platform-spec.ts (hunk 3 above)
Comment (bobbyg603) [reply spec/support/platform-spec.ts:3]: Renamed to `platformSpec`

**Pull request merged**